tools/subtype-gen/testdata/complex_nested_and_or_test.golden.go

```diff
@@ -0,0 +1,54 @@
+// Code generated from rules.yaml. DO NOT EDIT.
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright Flow Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sema
+
+func checkSubTypeWithoutEquality_gen(subType Type, superType Type) bool {
+	if subType == NeverType {
+		return true
+	}
+
+	switch superType {
+	case ComplexType:
+		switch typedSubType := subType.(type) {
+		case *IntersectionType:
+			switch typedSubType.LegacyType {
+			case nil,
+				AnyResourceType,
+				AnyStructType,
+				AnyType:
+				return false
+			}
```

Comment:
Would the generated code of the `equals` + `oneOf` which is _not_ contained in a `not` like here, result in an additional `default` case which returns false?

Reply:
Great! Thank you for the explanation 👍